src/lazy/raw_value_ref.rs

```diff
@@ -18,7 +19,7 @@ pub enum RawValueRef<'data, D: LazyDecoder<'data>> {
     Timestamp(Timestamp),
     String(StrRef<'data>),
     Symbol(RawSymbolTokenRef<'data>),
-    Blob(&'data [u8]),
+    Blob(BytesRef<'data>),
```

Comment:
🗺️ `RawValueRef` now returns a `BytesRef` instead of a `&[u8]` so it has the option to allocate a `Vec<u8>` when the input encoding is base64 text. The binary reader can still return a slice of the input buffer.